gadgets/src/bits/boolean_input.rs

```diff
@@ -95,20 +95,49 @@ impl<F: PrimeField, CF: PrimeField> AllocGadget<Vec<F>, CF> for BooleanInputGadg
     ) -> Result<Self, SynthesisError> {
         let obj = value_gen()?;
 
-        // convert the elements into booleans (little-endian)
-        let mut res = Vec::<Vec<Boolean>>::new();
-        for (i, elem) in obj.borrow().iter().enumerate() {
+        // Step 1: obtain the bits of the F field elements
+        let mut src_bits = Vec::<bool>::new();
+        for elem in obj.borrow().iter() {
             let mut bits = elem.to_repr().to_bits_le();
             bits.truncate(F::size_in_bits());
+            for _ in bits.len()..F::size_in_bits() {
+                bits.push(false);
+            }
```

Comment:
```suggestion
            bits.resize(F::size_in_bits(), false);
```
(untested, but should work)